code/modules/spells/spell_types/genetic.dm

```diff
@@ -28,7 +28,8 @@
 		for(var/A in traits)
 			ADD_TRAIT(target, A, GENETICS_SPELL)
 		active_on += target
-		addtimer(CALLBACK(src, .proc/remove, target), duration)
+		if(duration < charge_max)
+			addtimer(CALLBACK(src, .proc/remove, target), duration)

```

Comment:
If you want to fix a bug, add TIMER_OVERRIDE|TIMER_UNIQUE to this. Currently if the cooldown is shorter than the duration, there are some issues with it toggling off early.

Reply:
How would I do that, exactly?

Reply:
I believe it goes in the third arg position of addtimer()